resources/containers/field-modal.tsx

```diff
@@ -0,0 +1,129 @@
+import React, { useState, useEffect } from "react";
+import Modal from "react-modal";
+import styles from "./styles/field-modal.module.css";
+
+interface FieldModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  onSave: (data: any) => void;
+  fieldType: string;
+}
+
+const FieldModal: React.FC<FieldModalProps> = ({
+  isOpen,
+  onClose,
+  onSave,
+  fieldType,
+}) => {
+  const [selectedOptions, setSelectedOptions] = useState<string[]>([]);
+  const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+
+  const options = ["number", "text"];
+
+  useEffect(() => {
+    setSelectedOptions([]);
+  }, [isOpen]);
+
+  const handleSave = () => {
+    if (fieldType === "number") {
+      onSave({ maxValue: parseInt(selectedOptions[0] || "0") });
+    } else if (fieldType === "multiselect") {
+      onSave({ options: selectedOptions });
+    }
+    onClose();
+  };
+
+  const handleOptionSelect = (option: string) => {
+    if (!selectedOptions.includes(option)) {
+      setSelectedOptions([...selectedOptions, option]);
+    }
+    setIsDropdownOpen(false);
+  };
+
+  const handleRemoveOption = (optionToRemove: string) => {
+    setSelectedOptions(
+      selectedOptions.filter((option) => option !== optionToRemove),
+    );
+  };
+
+  return (
+    <Modal
+      isOpen={isOpen}
+      onRequestClose={onClose}
+      className={styles.modalContent}
+      overlayClassName={styles.modalOverlay}
+    >
+      <div className={`items-center ${styles.title}`}>
+        <h2 className="text-xl font-semibold">Edit Config</h2>
+      </div>
+      {fieldType === "number" ? (
```

Comment:
If else won't work here, this is a basic case, we can have 10 types with different convinations. 

We need a has map for the columns 

````
inputs = {
  "number":  (),
  "select", "multiselect": ()
}